apps/vs-agent/src/controllers/admin/verifiable/TrustService.ts

```diff
@@ -97,13 +97,19 @@ export class TrustService {
         verifiableCredential: [credential],
       })
       const integrityData = generateDigestSRI(JSON.stringify(credential.credentialSubject))
-      didRecord.didDocument?.service?.push(
-        new DidDocumentService({
-          id: `${agent.did}#vpr-${schemaKey}-c-vp`,
-          serviceEndpoint,
-          type: 'LinkedVerifiablePresentation',
-        }),
-      )
+      const newService = new DidDocumentService({
+        id: `${agent.did}#vpr-${schemaKey}-c-vp`,
+        serviceEndpoint,
+        type: 'LinkedVerifiablePresentation',
+      })
+
+      // TODO: review if possible allow duplicate id or endpoint
+      if (didRecord.didDocument) {
+        didRecord.didDocument.service = (didRecord.didDocument?.service ?? []).filter(
+          s => s.id !== newService.id && s.serviceEndpoint !== newService.serviceEndpoint,
```

Comment:
I don't get why you are looking also by serviceEndpoint. This makes it possible to have two services with the same id and that would be an issue since it is not compatible with the DID Core spec and may result in issues when resolving the DID Document:

> The value of the id property MUST be a [URI](https://www.w3.org/TR/did-1.0/#dfn-uri) conforming to [[RFC3986](https://www.w3.org/TR/did-1.0/#bib-rfc3986)]. A [conforming producer](https://www.w3.org/TR/did-1.0/#dfn-conforming-producer) MUST NOT produce multiple service entries with the same id. A [conforming consumer](https://www.w3.org/TR/did-1.0/#dfn-conforming-consumer) MUST produce an error if it detects multiple service entries with the same id.